src/edu/csus/ecs/pc2/clics/API202306/RunService.java

```diff
@@ -0,0 +1,145 @@
+                if (run.getElapsedMS() / 1000 > freezeTime) {
+                    continue;
+                }
+            }
+            if(run.isJudged() && !run.getJudgementRecord().isPreliminaryJudgement()) {
+                for(RunTestCase testCase: run.getRunTestCases()) {
+                    tclist.add(new CLICSTestCase(model, testCase));
+                }
+            }
+        }
+        try {
+            ObjectMapper mapper = JSONUtilities.getObjectMapper();
+            String json = mapper.writeValueAsString(tclist);
+            return Response.ok(json, MediaType.APPLICATION_JSON).build();
+        } catch (Exception e) {
+            return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Error creating JSON for Runs testcases " + e.getMessage()).build();
+        }
+    }
+
+    /**
+     * Returns a representation of the specified test case in the specified contest in JSON format. The returned value is compliant with 2023-06 API.
+     * 
+     * @param sc User's info
+     * @param contestId The contest
+     * @param runId The run of interest
+     * @return response
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{runId}/")
+    public Response getRun(@Context SecurityContext sc, @PathParam("contestId") String contestId, @PathParam("runId") String runId) {
+
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == true) {
+            long freezeTime = Utilities.getFreezeTime(model);
+    
+            for (Run run: model.getRuns()) {
+                // If not admin or judge, can not see runs after freeze time
+                if (!sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) && !sc.isUserInRole(WebServer.WEBAPI_ROLE_JUDGE)) {
+                    // if run is after scoreboard freeze, do not return info for it
+                    if (run.getElapsedMS() / 1000 > freezeTime) {
+                        continue;
+                    }
+                }
+                if(run.isJudged() && !run.getJudgementRecord().isPreliminaryJudgement()) {
+                    for(RunTestCase testCase: run.getRunTestCases()) {
+                        if (testCase.getElementId().toString().equals(runId)) {
+                            return Response.ok(new CLICSTestCase(model, testCase).toJSON(), MediaType.APPLICATION_JSON).build();
+                        }
+                    }
+                }
+            }
+        }
+        return Response.status(Response.Status.NOT_FOUND).build();
+    }
+    
+    /**
+     * Retrieve access information about this endpoint for the supplied user's security context
+     * 
+     * @param sc User's security information
+     * @return CLICSEndpoint object if the user can access this endpoint's properties, null otherwise
+     */
+    public static CLICSEndpoint getEndpointProperties(SecurityContext sc) {
+        return(new CLICSEndpoint("runs", JSONUtilities.getJsonProperties(CLICSTestCase.class)));
```

Comment:
Same code, same comment :)  Maybe it's me?

Reply:
My answer here is different than my other responses to the same question because this is a bug.  According to the IAP, only Analysts and Admins/Judges can see Runs (RunTestCases).  Everyone else can't see them.  I fixed this in the next push.